### PR TITLE
Add a static flag in EF that will be set when code is being executed for design-time discovery

### LINKED ISSUE #27306: Add a static flag in EF that will be set when code is being executed for design-time discovery
I have strange behavior with my EF Core tools in developer powershell in VS2019.

I am creating migration with dotnet ef migrations add VisibleLink3 -p .\src\Only.Portal.Data\ -s .\src\Only.Portal.Web command.

And it is starting my app, but previously it didn't.method to apply last migrations. Which causes dotnet ef migrations remove fully broken, because when using it, it firstly start app and Migrate().

Then I have a message:

The migration '20220128090939_VisibleLink3' has already been applied to the database. Revert it and try again. If the migration has been applied to other databases, consider reverting its changes using a new migration instead..

Looks like dead end loop.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #27306.

### CODE REVIEW

**Comment by ajcvickers** [comment]
@bricelam Is there a better way to test this?

File: src/EFCore.Design/Design/OperationExecutor.cs [hunk 1]
```diff
@@ -697,6 +697,7 @@ public abstract class OperationBase : MarshalByRefObject
         /// <param name="resultHandler">The <see cref="IOperationResultHandler" />.</param>
         protected OperationBase(IOperationResultHandler resultHandler)
         {
+            EF.IsDesignTime = true;
```
Comment (bricelam) [inline src/EFCore.Design/Design/OperationExecutor.cs:700]: Hmm, maybe move this to `Execute`.

I wonder if we should also set it back to `false` afterwards incase the process/AppDomain is reused. Probably not very important.

**Review by bricelam — COMMENTED** [review]
We should also set it in `DbContextActivator.CreateInstance`.

**Comment by bricelam** [comment]
> Is there a better way to test this?

This is exactly how I'd do it.

**Comment by ajcvickers** [comment]
@bricelam New version up.

**Review by bricelam — CHANGES_REQUESTED** [review]
We should also set it in `DbContextActivator.CreateInstance`. That's is what [ASP.NET Scaffolding uses](https://github.com/dotnet/Scaffolding/blob/main/src/Scaffolding/VS.Web.CG.EFCore/EntityFrameworkModelProcessor.cs#L562) at design-time.

**Review by bricelam — APPROVED** [review]

**Pull request merged**